src/Middleware/ExceptionHandlingMiddleware.php

```diff
@@ -0,0 +1,58 @@
+<?php declare(strict_types=1);
+namespace LORIS\Middleware;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \Psr\Http\Server\MiddlewareInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * ExceptionHandlingMiddleware is a type of Middleware which safely handles
+ * exceptions that are not caught earlier by other code. It makes sure that
+ * even when something unexpected happens that we can serve a decorated page
+ * to the user.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class ExceptionHandlingMiddleware implements MiddlewareInterface, MiddlewareChainer
+{
+    use MiddlewareChainerMixin;
+
+    /**
+     * Attempts a request and catches stray exceptions. When an exception occurs
+     * hand this off to another Middleware to decoreate the page.
+     *
+     * @param ServerRequestInterface  $request The incoming request
+     * @param RequestHandlerInterface $handler The handler to delegate
+     *                                         valid requests to
+     *
+     * @return ResponseInterface the PSR15 response.
+     */
+    public function process(
+        ServerRequestInterface $request,
+        RequestHandlerInterface $handler
+    ) : ResponseInterface {
+        try {
+            return $this->next->process($request, $handler);
+        } catch (\NotFound $e) {
+            error_log($e->getMessage() . $e->getTraceAsString());
+            $status = 404;
+        } catch (\Exception $e) {
+            error_log($e->getMessage() . $e->getTraceAsString());
+            $status = 500;
+        }
+
+        // Decorate the request.
+        return (new \LORIS\Middleware\PageDecorationMiddleware(
+            \NDB_Factory::singleton()->user()
```

Comment:
This should be coming from the request object 'user' attribute which is populated by the BaseRouter.

Reply:
This is still not done.

Reply:
I think this had similar issues to the Anonymous User call above. When I changed it there were cases where the 'user' attribute was null.

Reply:
@driusan Can you follow up on this point and the above point?

Reply:
@driusan bump

Reply:
There should never be cases where it's null, it's added unconditionally from the BaseRouter

Reply:
I just tested again. It works as long as the Anonymous User check is there. Otherwise this error occurs when a user is not logged in:

Uncaught TypeError: Argument 1 passed to LORIS\\Middleware\\PageDecorationMiddleware::__construct() must be an instance of User, null given, called in /var/www/loris/src/Middleware/ExceptionHandlingMiddleware.php on line 47 and defined in /var/www/loris/src/Middleware/PageDecorationMiddleware.php:18